dracut/28immucore/generator.sh

```diff
@@ -8,25 +8,20 @@ GENERATOR_DIR="$2"
 [ -z "$GENERATOR_DIR" ] && exit 1
 [ -d "$GENERATOR_DIR" ] || mkdir "$GENERATOR_DIR"
 
-
-## GENERATE SYSROOT
-cos_img=$(getarg cos-img/filename=)
-[ -z "${cos_img}" ] && exit 0
-
-# This is necessary because otherwise systemd-fstab-generator will se the cmdline with the root=LABEL=X stanza and
-# say, hey this is the ROOT where we need to boot! so it auto creates a sysroot.mount with the content of the value
-# passed in the cmdline. But because we usually pass the label of the img (COS_ACTIVE) it will create the wrong mount
-# service and be stuck in there forever.
-# by generating it ourselves we get the sysroot.mount into the generators.early dir, which tells systemd to not generate it
-# as it already exists and the rest is history
+# Add a timeout to the sysroot so it waits a bit for immucore to mount it properly
+mkdir -p "$GENERATOR_DIR"/sysroot.mount.d
 {
     echo "[Unit]"
-    echo "Before=initrd-root-fs.target"
-    echo "DefaultDependencies=no"
     echo "[Mount]"
```

Comment:
is then reading the fstab? what will try to mount systemd then here? ( mmm guess it's a no-op)

Reply:
The original sysroot comes from dracut parsing the cmdline looking for the `root` stanza. That generates the sysroot.mount in all cases.but systemd is smart enough that if it sees something being mounted on /sysroot it updates the original sysroot.mount with the new values and marks it as success.

As we mount in /sysroot we take advantage of that and can ignore creating the mount ourselves as it will be marked as mounted as soon as immucore mounts the sysroot.

Just less code to create ourselves.

And the timeout added is just so sysroot doesn't fail immediately on CD boot.

Reply:
oh right true, thanks for the clarification!